spring-cloud-stream/src/main/java/org/springframework/cloud/stream/function/FunctionConfiguration.java

```diff
@@ -487,11 +487,8 @@ private void bindFunctionToDestinations(BindableProxyFactory bindableProxyFactor
 					}
 					MessageChannel inputChannel = null;
 					final String reactive = environment.getProperty("spring.cloud.stream.reactive");
-					if (StringUtils.hasText(reactive)) {
-						boolean reactiveFn = Boolean.parseBoolean(reactive);
-						if (reactiveFn) {
-							inputChannel = this.applicationContext.getBean(inputBindingName, FluxMessageChannel.class);
-						}
+					if (Boolean.parseBoolean(reactive)) {
+						inputChannel = this.applicationContext.getBean(inputBindingName, FluxMessageChannel.class);
 					}
 					else {
```

Comment:
I wonder why do we need this `if...else` at all?
Why can't we just have `this.applicationContext.getBean(inputBindingName, MessageChannel.class);`?
The next ` IntegrationReactiveUtils.messageChannelToFlux()` is good to deal with any type of channels and you always get a good `Flux` back from there:
```
	public static <T> Flux<Message<T>> messageChannelToFlux(MessageChannel messageChannel) {
		if (messageChannel instanceof Publisher) {
			return Flux.from((Publisher<Message<T>>) messageChannel);
		}
		else if (messageChannel instanceof SubscribableChannel) {
			return adaptSubscribableChannelToPublisher((SubscribableChannel) messageChannel);
		}
		else if (messageChannel instanceof PollableChannel) {
			return messageSourceToFlux(() -> (Message<T>) ((PollableChannel) messageChannel).receive(0));
		}
		else {
			throw new IllegalArgumentException("The 'messageChannel' must be an instance of Publisher, " +
					"SubscribableChannel or PollableChannel, not: " + messageChannel);
		}
	}
```